test/skimNtuple_HHbtag.cpp

```diff
@@ -3361,102 +3422,108 @@ int main (int argc, char** argv)
-						  SFtau_Data_down[idm] = tauTrgSF_etau->getEfficiencyData(tlv_secondLepton.Pt(), DM2, -1);
-						  SFtau_MC_up[idm]     = tauTrgSF_etau->getEfficiencyMC  (tlv_secondLepton.Pt(), DM2, 1);
-						  SFtau_MC_down[idm]   = tauTrgSF_etau->getEfficiencyMC  (tlv_secondLepton.Pt(), DM2, -1);
+				  double Eff_Data = passSingle * Eff_SL_ele_Data - passCross * passSingle * std::min(Eff_cross_ele_Data, Eff_SL_ele_Data) * Eff_cross_tau_Data + passCross * Eff_cross_ele_Data * Eff_cross_tau_Data;
+				  double Eff_MC   = passSingle * Eff_SL_ele_MC   - passCross * passSingle * std::min(Eff_cross_ele_MC  , Eff_SL_ele_MC)   * Eff_cross_tau_MC   + passCross * Eff_cross_ele_MC   * Eff_cross_tau_MC;
+
+				  // Compute uncertainties (separate for SingleElectron and ele leg of cross trigger)
+				  double Err_Data_SL_ele = passSingle * Eff_SL_ele_Data_Err - passCross * passSingle * (Eff_cross_ele_Data > Eff_SL_ele_Data) * Eff_SL_ele_Data_Err * Eff_cross_tau_Data;
+				  double Err_MC_SL_ele   = passSingle * Eff_SL_ele_MC_Err   - passCross * passSingle * (Eff_cross_ele_MC   > Eff_SL_ele_MC)   * Eff_SL_ele_MC_Err   * Eff_cross_tau_MC;
+
+				  double trigSF_SL_ele_err = 0.;
+				  if(passSingle) {
+					trigSF_SL_ele_err = eTrgSF->get_ScaleFactorError(Eff_Data, Eff_MC, Err_Data_SL_ele, Err_MC_SL_ele);
+				  }
+
+				  double Err_Data_cross_ele = - passCross * passSingle * (Eff_cross_ele_Data <= Eff_SL_ele_Data) * Eff_cross_ele_Data_Err * Eff_cross_tau_Data + passCross * Eff_cross_ele_Data_Err * Eff_cross_tau_Data;
+				  double Err_MC_cross_ele   = - passCross * passSingle * (Eff_cross_ele_MC   <= Eff_SL_ele_MC)   * Eff_cross_ele_MC_Err   * Eff_cross_tau_MC   + passCross * Eff_cross_ele_MC_Err   * Eff_cross_tau_MC;
+
+				  double trigSF_cross_ele_err = 0.;
+
+				  // if single lepton trigger and cross trigger are passed and the cross trigger is less efficient than
+				  // the single lepton trigger, the "combined" efficiency reduces to the single lepton trigger efficiency
+				  // and the uncertainty on the cross trigger leg is 0, if this is the case in both Data and MC also the
+				  // uncertaintiy on the trigger scale factor is 0
+				  if(passCross and !(passSingle and (Eff_cross_ele_Data <= Eff_SL_ele_Data or Eff_cross_ele_MC <= Eff_SL_ele_MC))) {
+					trigSF_cross_ele_err = eTrgSF->get_ScaleFactorError(Eff_Data, Eff_MC, Err_Data_cross_ele, Err_MC_cross_ele);
+				  }
+
+				  double Err_Data_ele = Err_Data_SL_ele + Err_Data_cross_ele;
+				  double Err_MC_ele   = Err_MC_SL_ele   + Err_MC_cross_ele;
+				  double trigSF_ele_err   = eTrgSF->get_ScaleFactorError(Eff_Data, Eff_MC, Err_Data_ele, Err_MC_ele);
+				  // for each DM, get the trigSF error if the tauh has the corresponding DM (and the cross trigger thresholds are passed), otherwise 0
+				  vector <double> trigSF_err    (N_tauhDM, 0.);
+				  if(passCross and ((Eff_cross_ele_Data > Eff_SL_ele_Data) or (Eff_cross_ele_MC > Eff_SL_ele_MC))){
```

Comment:
Just to be 100% sure: aren't you missing a ```passSingle``` term here, just like in other similar code lines?

Reply:
Yes I was, I've combined the two if statements now anyway